src/colmap/estimators/cost_functions.h

```diff
@@ -229,36 +305,44 @@ class RigReprojErrorCostFunction {
                             &residuals[1]);
     residuals[0] -= T(observed_x_);
     residuals[1] -= T(observed_y_);
+    if constexpr (CTYPE != CovarianceType::IDENTITY) {
+      ApplyCovariance<T, 2, CTYPE>(residuals, sqrt_information_point_);
+    }
     return true;
   }
 
  private:
   const double observed_x_;
   const double observed_y_;
+  const Eigen::Matrix2d sqrt_information_point_;
```

Comment:
```suggestion
  const Eigen::Vector2d point2D_;
  const Eigen::Matrix2d sqrt_info_point2D_;
```
nit: Maybe more readable?